weed/mount/weedfs_file_lseek.go

```diff
@@ -0,0 +1,110 @@
+package mount
+
+import (
+	"syscall"
+
+	"github.com/hanwen/go-fuse/v2/fuse"
+
+	"github.com/seaweedfs/seaweedfs/weed/filer"
+	"github.com/seaweedfs/seaweedfs/weed/glog"
+)
+
+// These are non-POSIX extensions
+const (
+	SEEK_DATA uint32 = 3 // seek to next data after the offset
+	SEEK_HOLE uint32 = 4 // seek to next hole after the offset
+	ENXIO            = fuse.Status(syscall.ENXIO)
+)
+
+// Lseek finds next data or hole segments after the specified offset
+// See https://man7.org/linux/man-pages/man2/lseek.2.html
+func (wfs *WFS) Lseek(cancel <-chan struct{}, in *fuse.LseekIn, out *fuse.LseekOut) fuse.Status {
+	// not a documented feature
+	if in.Padding != 0 {
+		return fuse.EINVAL
+	}
+
+	if in.Whence != SEEK_DATA && in.Whence != SEEK_HOLE {
```

Comment:
How about supporting POSIX first?

Reply:
This is not possible in FUSE. We can only implement seeking data or holes. Everything else in handled by the kernel. Note for example the `Read` and `Write` calls we receive. It already includes the `Offset` parameter, which is given by the kernel.

Reply:
this means any code that calls `lseek()` with common POSIX whence would fail?

Reply:
No, the kernel handles `SEEK_SET`, `SEEK_CUR` and `SEEK_END`, just like now.

Reply:
In fact, FUSE is not even notified about POSIX whences, we even have no possibility at all to keep track of the current seek offset, it's all in the kernel.

Reply:
Some further references:
https://github.com/libfuse/libfuse/pull/457/files#diff-59f547e8f69480e75edc12102f19fa1dc4891aaff33633b96235e5990cc89698R1237
https://patchwork.kernel.org/project/linux-fsdevel/patch/1452835635-14440-1-git-send-email-ravishankar@redhat.com/
https://patchwork.kernel.org/project/qemu-devel/patch/20200922104932.46384-7-mreitz@redhat.com/

Reply:
thanks for the docs! Good for learning!

Reply:
In `linux-fsdevel` you can also see that only `SEEK_DATA` and `SEEK_HOLE` is delegated to libfuse. In `qemu-devel` you can see the same check:
```
if (whence == SEEK_DATA || whence == SEEK_HOLE) {
    fuse_reply_lseek(req, offset);
} else {
    fuse_reply_err(req, EINVAL);
}
```